### ISSUE TITLE
Model returns superior result with learning_rate 100

### ISSUE THREAD

**Issue by mhschel** [microsoft/LightGBM#2723]
## Environment info

Operating System:
Debian Buster

CPU/GPU model:
Rizen3950X, no GPU used

C++/Python/R version:
Reproduced with both Python and R API's.

LightGBM version or commit hash:
2.3.2

## Reproducible examples
I have a tiny dataset of 200 samples. When I fit a gbm with the R `gbm` package, it reduces train error significantly. LightGBM behaves differently. Fitting up to 2000 trees only decreases train error by 3% when `learning_rate` is set to 1. I have experimented with many parameters and the only parameter that seems to have a big positive effect is by setting learning_rate to 10 or 100. Setting `learning_rate` to 100 reduces train_error by 75%. I have never seen learning rates used above 1.

How it this possible? Am I missing something important about LightGBM?

## Steps to reproduce

1. Download the data:
[reprocdata.zip](https://github.com/microsoft/LightGBM/files/4140121/reprocdata.zip)
2. Run code below
3. Change `learning_rate` to 100 to see big performance increase.

> import lightgbm as lgb
> import pandas as pd
> 
> lgb_data = pd.read_csv('jhtest2.csv')
> lgb_label = pd.read_csv('jh_test_ytr.csv').prices_train
> 
> lgb_train = lgb.Dataset(
>     lgb_data,
>     lgb_label,
>     categorical_feature=['street']
>     )
> 
> params = {
>         'objective': 'huber',
>         'min_data_in_leaf': 5,
>         'feature_fraction': 0.8,
>         'bagging_fraction': 0.8,
>         'nthreads': 4,
>         'learning_rate': 1,
>         'verbose': 2,
>     }
> 
> test_lgb = lgb.train(
>     params=params,
>     train_set=lgb_train,
>     valid_sets=[lgb_train],
>     num_boost_round=2000,
>     verbose_eval=100,
>     early_stopping_rounds=50,
>     categorical_feature=['street']
> )

**Comment by guolinke** [comment]
I think it is related to `huber`, which will force the gradients to constants if delta of label and score is too large.
